### PR TITLE
Sample code for joinSeedNodes programatically

### LINKED ISSUE #24041: Lack of documents about joining seed-nodes programmatically
I don't want to use the `application.conf` as document said.
The code I use:
```
System.setProperty("akka.actor.provider", "cluster");
System.setProperty("akka.remote.netty.tcp.hostname","127.0.0.1");
System.setProperty("akka.remote.netty.tcp.port","2551");

final ActorSystem system = ActorSystem.create("ClusterSystem");

Address address1 = new Address("akka.tcp", "ClusterSystem@127.0.0.1:2551");
List<Address> list = new LinkedList<>();
list.add(address1);

Cluster.get(system).joinSeedNodes(list);
```

Got the following log:
```
[WARN] [11/22/2017 18:04:18.336] [ClusterSystem-akka.actor.default-dispatcher-2] [akka.tcp://ClusterSystem@127.0.0.1:2551/system/cluster/core/daemon/joinSeedNodeProcess-1] Couldn't join seed nodes after [2] attempts, will try again. seed-nodes=[akka.tcp://ClusterSystem@127.0.0.1:2551]
[INFO] [11/22/2017 18:04:18.340] [ClusterSystem-akka.actor.default-dispatcher-15] [akka.cluster.Cluster(akka://ClusterSystem)] Cluster Node [akka.tcp://ClusterSystem@127.0.0.1:2551] - Received InitJoin message from [Actor[akka://ClusterSystem/system/cluster/core/daemon/joinSeedNodeProcess-1#-253852842]], but this node is not initialized yet
```

It seems like the cluster didn't form successfully.
what's going on and what is the right way to do.

### PR DESCRIPTION
**Pull request by richardimaoka**
Closes #24041

### CODE REVIEW

File: akka-docs/src/main/paradox/cluster-usage.md [hunk 1]
```diff
@@ -94,6 +94,8 @@ The source code of this sample can be found in the
   [Akka Cluster Bootstrap](https://developer.lightbend.com/docs/akka-management/current/bootstrap.html) module.
 @@@
 
+### Manually joining to seed nodes
```
Comment (ktoso) [inline akka-docs/src/main/paradox/cluster-usage.md]: Maybe call this Joining configured seed nodes; manually can mean many things

File: akka-docs/src/main/paradox/cluster-usage.md [hunk 2]
```diff
@@ -142,12 +138,28 @@ form a new cluster instead of joining remaining nodes of the existing cluster. T
 likely not desired and should be avoided by listing several nodes as seed nodes for redundancy
 and don't stop all of them at the same time.
 
+### Automatically joining to seed nodes with Cluster Bootstrap
+
+Instead of manually configuring seed nodes, which is useful in development or statically assigned node IPs, you may want
+to automate the discovery of seed nodes using your cloud providers or cluster orchestrator, or some other form of service
+discovery (such as managed DNS). The open source Akka Management library includes the
+[Cluster Bootstrap](https://developer.lightbend.com/docs/akka-management/current/bootstrap.html) module which handles
+just that. Please refer to its documentation for more details.
```
Comment (ktoso) [inline akka-docs/src/main/paradox/cluster-usage.md]: nice

**Review by ktoso — APPROVED** [review]
Great, thanks :)

File: akka-docs/src/main/paradox/cluster-usage.md (hunk 1 above)
Comment (richardimaoka) [reply akka-docs/src/main/paradox/cluster-usage.md]: Sure, updated.

File: akka-docs/src/test/java/jdocs/cluster/ClusterDocTest.java [hunk 3]
```diff
@@ -57,4 +61,13 @@ public void demonstrateDataCenter() {
     //#dcAccess
   }
 
+  // compile only
+  @SuppressWarnings("unused")
+  public void demonstrateJoinSeedNodes() {
+    //#join-seed-nodes
+    final Cluster cluster = Cluster.get(system);
```
Comment (raboof) [inline akka-docs/src/test/java/jdocs/cluster/ClusterDocTest.java]: Could we include the `Cluster` and `Address` imports to the snippet?

File: akka-docs/src/test/scala/docs/cluster/ClusterDocSpec.scala [hunk 4]
```diff
@@ -41,4 +41,14 @@ class ClusterDocSpec extends AkkaSpec(ClusterDocSpec.config) with CompileOnlySpe
     }
   }
 
+  "demonstrate programatic joining to seed nodes" in compileOnlySpec {
+    //#join-seed-nodes
+    import akka.actor.Address
```
Comment (raboof) [inline akka-docs/src/test/scala/docs/cluster/ClusterDocSpec.scala]: :+1:, also import `Cluster`?

**Review by raboof — COMMENTED** [review]
Nice! Would be nice to include the main imports in the snippet, after that LGTM!

**Review by patriknw — APPROVED** [review]
LGTM

File: akka-docs/src/test/scala/docs/cluster/ClusterDocSpec.scala (hunk 4 above)
Comment (richardimaoka) [reply akka-docs/src/test/scala/docs/cluster/ClusterDocSpec.scala]: Okay sure, added imports in the new commit https://github.com/akka/akka/pull/24982/commits/90ed27225697416fdc20597f0c96a172eb63ea9d

2018-04-29_21h01_46
2018-04-29_21h01_53

**Pull request merged**